ialirt_data_access/io.py

```diff
@@ -145,3 +145,66 @@ def download(filename: str, downloads_dir: Optional[Path] = None) -> Path:
             print(f"Successfully downloaded the file to: {destination}")
 
     return destination
+
+
+def algorithm_query(
```

Comment:
You are calling this `_db_query()` in the CLI. It'd be good to be consistent with one or the other. I don't think you are querying "algorithms", but db is also pretty general and you have different tables I think. Maybe `data_product_query` in both locations?